top.wake

```diff
@@ -47,19 +47,48 @@ global def bitstreamTarget board dut buildRoot =
 
 global def simTarget dut buildRoot simName waves =
   def dutTests = subscribe dutTests | filter (_.getDUTTestFilter dut)
-  map (singleSimTarget simName waves dut buildRoot) dutTests
-  | zip dutTests
-  | map getDUTTestName.editPairFirst
-  | aggregateTestResults
-  | println
+  def programNames = map (_.getDUTTestProgram.getTestProgramPlanName) dutTests
+  def simOutputs = map (singleSimOutput simName waves dut buildRoot) dutTests
+
+  def simCheckerOptions programName =
+    def filter = (\_ True)
+    def outputDir = buildRoot.getBuildTreeReportDir simName programName
+    makeSimCheckerOptions filter outputDir
+
+  def allPassed =
+    simOutputs
+    | zip programNames
```

Comment:
my only worry is using zip here. if simOutputs and programNames get out of sync in the future, this could be a problem.

And maybe add some comments